Setting up CI

j-hellenberg opened the pull request:
With a lot of trial and error, CI now finally seems to work.

I haven't noticed any difference between github actions and travis though; both seem to now work fine and produce the same results with the main difference being that travis reports coverage to coveralls.io while github actions does not.

When merged, the CI configuration and the README.md badges should now target the dev branch.

Note that I also had to change one line of source code to make all current tests succeed.

Closes #74.

j-hellenberg committed:
.yml for SmalltalkCI via Github Actions

j-hellenberg committed:
Delete .travis.yml

j-hellenberg committed:
Update and rename .yml to main.yml

j-hellenberg committed:
test

j-hellenberg committed:
Update main.yml

j-hellenberg committed:
test #2

j-hellenberg committed:
Rename main.yml to .github/workflows/main.yml

j-hellenberg committed:
Update main.yml

j-hellenberg committed:
Update main.yml

j-hellenberg committed:
Update main.yml

j-hellenberg committed:
Update main.yml

j-hellenberg committed:
Update baseline..st

j-hellenberg committed:
Changed build badge to match github action

Commit:
Replaced array>>sortBy: by sorted: since sortBy: is deprecated

j-hellenberg committed:
Test reenable travis

j-hellenberg committed:
Update .travis.yml

j-hellenberg committed:
Update README for travis

j-hellenberg committed:
Changes CI target to dev branch

frcroth commented:
Wow that's a lot of commits. Maybe squash?

frcroth commented on packages/RichTextEditing-Core.package/RichTextDocument.class/instance/sortRunsByPriority.st:
```diff
@@ -5,6 +5,6 @@ sortRunsByPriority
 		structureCopy := attributes copy select: [:attribute | attribute isTextStructure].
 		noStructureCopy :=  attributes copy reject: [:attribute | attribute isTextStructure].
-		structureCopy := structureCopy copy sortBy: [:a :b | 
+		structureCopy := structureCopy sorted: [:a :b | 
```
Why?

j-hellenberg replied:
Because sortBy: is deprecated and the message is therefore not understood in the CI images any more.

frcroth reviewed (commented):
Seems alright, pls explain what you did

j-hellenberg commented:
> Wow that's a lot of commits. Maybe squash?

True. But all of them contain only minor changes, since when changing files on Github directly, every change produces an separate commit.
Squashing would cerainly make sense here, but I don't know how to do so on Github and for some weird reason I am unable to find the files outside of my squeak-image on local...

j-hellenberg commented:
> Seems alright, pls explain what you did

In overview:
- Created configuration file for github action (now targeting dev branch)
- Fixed dependency to MorphicTestingFramework
- Updated badges in README.md
- All tests are currently succeeding due to the small fix you mentioned above

frcroth committed:
Merge branch 'dev' into test-ci

frcroth commented:
Squashing can be done online, when the pull request is merged, by clicking the arrow next to "Merge pull request"

sirkrypt0 commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# RichTextEditing [![Build Status](https://travis-ci.org/hpi-swa-teaching/RichTextEditing.svg?branch=master)](https://travis-ci.org/hpi-swa-teaching/RichTextEditing)[![Coverage Status](https://coveralls.io/repos/github/hpi-swa-teaching/RichTextEditing/badge.svg?branch=master)](https://coveralls.io/github/hpi-swa-teaching/RichTextEditing?branch=master)
+# RichTextEditing ![CI](https://github.com/hpi-swa-teaching/RichTextEditing/workflows/CI/badge.svg)[![Build Status](https://travis-ci.org/hpi-swa-teaching/RichTextEditing.svg?branch=dev)](https://travis-ci.org/hpi-swa-teaching/RichTextEditing)[![Coverage Status](https://coveralls.io/repos/github/hpi-swa-teaching/RichTextEditing/badge.svg?branch=master)](https://coveralls.io/github/hpi-swa-teachin ...
```
I think we don’t need travis anymore, so the badge can surely be removed

sirkrypt0 commented on .travis.yml:
```diff
@@ -6,5 +6,5 @@ os:
-  - Squeak-5.2
-  - Squeak-5.1
+  - Squeak64-trunk
+  - Squeak64-5.3
```
I think we just remove travis at this point, as we have GitHub actions

j-hellenberg replied:
Not quite sure about this, since without travis, coverage is no longer reported.
Travis is actually succeeding, the badge only shows it as failing because it is already targeting the dev branch with the old travis configuration. Does not hurt to keep it, I guess.

sirkrypt0 replied:
oh, github action has no coverage report?

j-hellenberg replied:
Seems like it. The smalltalkci readme does only mention coverage for Travis or AppVeyor and the test-ci branch only appearend on coveralls after I reenabled travis.

sirkrypt0 commented:
It’s better to do a rebase when pulling the changes from dev branch to avoid merge commits

sirkrypt0 commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,27 @@
+# events but only for the master branch
+on:
+  push:
+    branches: [ dev ]
```
The CI should be running on each push on each branch. This allows for test driven development and warns you if you pushed something that breaks a test etc. If you run the CI only on pull requests, you might already have done a lot and it's unclear on which commit it exactly got wrong.

j-hellenberg committed:
CI should always run

j-hellenberg committed:
Removing windows from ci configuration

Since the tests seem to be not run there
